ReactCommon/hermes/executor/HermesExecutorFactory.cpp

```diff
@@ -221,6 +221,12 @@ std::unique_ptr<JSExecutor> HermesExecutorFactory::createJSExecutor(
       decoratedRuntime, delegate, jsQueue, timeoutInvoker_, runtimeInstaller_);
 }
 
+::hermes::vm::RuntimeConfig HermesExecutorFactory::defaultRuntimeConfig() {
+  return ::hermes::vm::RuntimeConfig::Builder()
+      .withEnableSampleProfiling(true)
```

Comment:
This enables the sampling profiler for all RN users by default. I don't know RN enough, but how is the HermesExecutorFactory create? You need to pipe the options to that call, and not set it here by default...

Reply:
Yes, this would enable it by default, which I think is reasonable. Do you know if there’s any overhead in enabling it? Should we do it only for debug builds?

It would still be possible to change by passing a hermes config when creating the HermesExecutorFactory. This is done in the user’s AppDelegate.mm for iOS (https://github.com/facebook/react-native/blob/main/packages/rn-tester/RNTester/AppDelegate.mm#L195) and Android also supports passing in a JavaScriptExecutor factory (https://github.com/facebook/react-native/blob/b6f7689d701d0409c23ab364356aeb95710c20fa/ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManagerBuilder.java#L87), although the Java binding doesn't seem to expose any options except heap size (https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/ReactAndroid/src/main/java/com/facebook/hermes/reactexecutor/RuntimeConfig.java). The android implementation also seems to enable sample profiler by default in a way that's not really configurable (https://github.com/facebook/react-native/blob/1bf84a63d83179d2676ef50eedf6bdd81e031b43/ReactAndroid/src/main/java/com/facebook/hermes/reactexecutor/OnLoad.cpp#L50).

Reply:
The problem is that the profiler machinery is thread-affined. To be honest we tried to change this several times but we couldn't find a solution :(

I understand your frustration -- we REALLY want to enable usage of the sampling profiler more broadly, but enabling it by default for all users could lead to usability issues -- incorrect stacks, runtime crashes etc.

I think at this point I'll defer to the RN folks on what they want to do. @cortinico?

Reply:
registerForProfiling/unregisterForProfiling must be invoked in the thread running JS (i.e., the "VM" thread), so the VM will be stopped by necessity.

Is there a way to enable this for iOS in a similar manner than it is currently done for Android?

Reply:
On iOS there's not much abstraction around HermesExecutorFactory, so we'd need to pass a hermes config when creating it.

Newer version of the app template actually abstract this away here https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/React/AppSetup/RCTAppSetupUtils.mm#L115 so we could use that to pass the hermes runtime config with profiler enabled.

Reply:
It seems that for now, the best solution would be to always set ```withEnableSampleProfiling``` in HermesExecutorFactory::createJSExecutor() to ensure all platforms can be profiled.

With that being said, someone from RN should have the final say here as to how they want this done. @cortinico 

@janicduplessis thank you for bringing this up to our attention! and for following through on all comments.

Reply:
From what I can tell, the sampling profiler has been enabled for all users before, and still is for android. It is no longer enabled by default since a recent change in hermes.

I haven't tried to see if it works, but would a better solution be to enable profiler when devtools connect?

Reply:
I imagine it would be possible to run that code on the JS VM thread, not sure if some other methods already do that.

Reply:
What's the consensus here. I would like to drive this forward and try to merge it. @jpporto I believe this is fine on RN's end. Are we fine merging it?